chia/solver/solver.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import asyncio
+import contextlib
+import logging
+from collections.abc import AsyncIterator
+from concurrent.futures.thread import ThreadPoolExecutor
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, ClassVar, Optional, cast
+
+from chia_rs import ConsensusConstants
+
+from chia.protocols.outbound_message import NodeType
+from chia.protocols.solver_protocol import SolverInfo
+from chia.rpc.rpc_server import StateChangedProtocol, default_get_connections
+from chia.server.server import ChiaServer
+from chia.server.ws_connection import WSChiaConnection
+
+log = logging.getLogger(__name__)
+
+
+class Solver:
+    if TYPE_CHECKING:
+        from chia.rpc.rpc_server import RpcServiceProtocol
+
+        _protocol_check: ClassVar[RpcServiceProtocol] = cast("Solver", None)
+
+    root_path: Path
+    _server: Optional[ChiaServer]
+    _shut_down: bool
+    started: bool = False
+    executor: ThreadPoolExecutor
+    state_changed_callback: Optional[StateChangedProtocol] = None
+    constants: ConsensusConstants
+    event_loop: asyncio.events.AbstractEventLoop
+
+    @property
+    def server(self) -> ChiaServer:
+        if self._server is None:
+            raise RuntimeError("server not assigned")
+
+        return self._server
+
+    def __init__(self, root_path: Path, config: dict[str, Any], constants: ConsensusConstants):
+        self.log = log
+        self.root_path = root_path
+        self._shut_down = False
+        num_threads = config["num_threads"]
+        self.log.info(f"Initializing solver with {num_threads} threads")
+        self.executor = ThreadPoolExecutor(max_workers=num_threads, thread_name_prefix="solver-")
+        self._server = None
+        self.constants = constants
+        self.state_changed_callback: Optional[StateChangedProtocol] = None
+        self.log.info("Solver initialization complete")
+
+    @contextlib.asynccontextmanager
+    async def manage(self) -> AsyncIterator[None]:
+        try:
+            self.log.info("Starting solver service")
+            self.started = True
+            self.log.info("Solver service started successfully")
+            yield
+        finally:
+            self.log.info("Shutting down solver service")
+            self._shut_down = True
+            self.executor.shutdown(wait=True)
+            self.log.info("Solver service shutdown complete")
+
+    def solve(self, info: SolverInfo) -> Optional[bytes]:
+        self.log.debug(f"Solve request: quality={info.partial_proof.hex()}")
+        # TODO todo_v2_plots implement actualy calling the solver
```

Comment:
I think this is all we need.
```suggestion
    def solve(self, partial_proof: bytes) -> Optional[bytes]:
        self.log.debug(f"Solve request: quality={partial_proof.hex()}")
        # TODO todo_v2_plots implement actualy calling the solver
```